### PR TITLE
Schedule presenter

### PR DESCRIPTION
**Pull request by Saadnajmi**
Working Presenter for the Schedule. Some issues that need to be addressed:

-Formatting DateTimeOffset objects to display on textviews nicely
-Not using FakeData(I don't know the actual REST URL sooooooo
-Making the DetailedEventActivity Layout look nicer (not a priority

-Subclassing Android Application to make sure we only have one HTTP Client (I plan to do that on  a separate PR since it's kinda tanget)

### CODE REVIEW

File: mobile/Client/Client.Common/ScheduleScreen/SchedulePresenter.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            View.ShowMessage("Subscribed for notifications for this game!");
+            // TODO: What the message says
+        }
+
+        public void OnClickRow(Event row)
+        {
+            View.OpenEvent(row);
+        }
+
+		private async Task UpdateFromServer()
```
Comment (zuhairp) [inline mobile/Client/Client.Common/ScheduleScreen/SchedulePresenter.cs]: Is this GitHub being weird or is the spacing actually like that?

File: mobile/Client/Client.Droid/Screens/ScheduleFragment.cs [hunk 2]
```diff
@@ -8,18 +8,44 @@
...
-            // Create your fragment here
+            Presenter = new SchedulePresenter(new ShowdownRESTClient());
+            Presenter.TakeView(this);
+
+            Adapter = new ScheduleAdapter()
+            {
+                //Events = new List<Event>()
+                Events = MakeFakeData()
+            };
+            Adapter.ItemClick += OnItemClick;
```
Comment (zuhairp) [inline mobile/Client/Client.Droid/Screens/ScheduleFragment.cs]: Does this work now?

File: mobile/Client/Client.Droid/Screens/ScheduleFragment.cs [hunk 3]
```diff
@@ -8,18 +8,44 @@
...
             base.OnCreate(savedInstanceState);
 
-            // Create your fragment here
+            Presenter = new SchedulePresenter(new ShowdownRESTClient());
+            Presenter.TakeView(this);
+
+            Adapter = new ScheduleAdapter()
+            {
+                //Events = new List<Event>()
+                Events = MakeFakeData()
```
Comment (zuhairp) [inline mobile/Client/Client.Droid/Screens/ScheduleFragment.cs]: Get this hooked up to backend before merging.

File: mobile/Client/Client.Droid/Screens/ScheduleFragment.cs [hunk 4]
```diff
@@ -28,24 +54,69 @@ public override View OnCreateView(LayoutInflater inflater, ViewGroup container,
...
             ScheduleView.SetLayoutManager(new LinearLayoutManager(this.Activity));
-
-            adapter.ItemClick += OnItemClick;
+            ScheduleView.SetAdapter(Adapter);
 
             return view;
         }
 
-        // Handler for the item click event:
+        // Handler for the item click event. This should be moved to the Presenter Methods instead
```
Comment (zuhairp) [inline mobile/Client/Client.Droid/Screens/ScheduleFragment.cs]: What does this comment mean?

File: mobile/Client/Client.Droid/Screens/ScheduleFragment.cs (hunk 2 above)
Comment (zuhairp) [inline mobile/Client/Client.Droid/Screens/ScheduleFragment.cs]: Also, you could just do 
```C#
Adapter.ItemClick += args => Presenter.OnRowClick(args.Event)
```

File: mobile/Client/Client.Droid/Screens/ScheduleFragment.cs [hunk 5]
```diff
@@ -28,24 +54,69 @@ public override View OnCreateView(LayoutInflater inflater, ViewGroup container,
...
         void OnItemClick(object sender, ScheduleAdapterClickEventArgs args)
         {
-            // Display a toast that briefly shows the enumeration of the selected photo:
-            int itemNumber = args.Position + 1;
-            Toast.MakeText(this.Activity, "This is event number " + itemNumber, ToastLength.Short).Show();
-            StartActivity(new Intent(this.Activity, typeof(DetailedEventActivity)));
+             Presenter.OnClickRow(args.Event);
```
Comment (zuhairp) [inline mobile/Client/Client.Droid/Screens/ScheduleFragment.cs]: If you use the anonymous handler style I mentioned, you can delete this method

File: mobile/Client/Client.Droid/Screens/ScheduleFragment.cs [hunk 6]
```diff
@@ -28,24 +54,69 @@ public override View OnCreateView(LayoutInflater inflater, ViewGroup container,
...
         }
+
+        void IScheduleView.OpenEvent(Event row)
+        {
+            var Intent = new Intent(this.Activity, typeof(DetailedEventActivity));
+            Intent.PutExtra("event", row.ToJSON());
+            StartActivity(Intent);
+        }
+
+        //REST Client isn't working right now, so this comes back
```
Comment (zuhairp) [inline mobile/Client/Client.Droid/Screens/ScheduleFragment.cs]: Get rid of this

File: mobile/Common/Common.Common/ShowdownRESTClient.cs [hunk 7]
```diff
@@ -17,7 +17,7 @@ public ShowdownRESTClient()
 			client = new HttpClient();
 		}
 
-		public async Task<List<Event>> GetScheduleAsync()
+		public async Task<List<Event>> GetSchedule()
```
Comment (zuhairp) [inline mobile/Common/Common.Common/ShowdownRESTClient.cs]: Why did you rename this?

File: mobile/Client/Client.Droid/Screens/ScheduleFragment.cs (hunk 4 above)
Comment (Saadnajmi) [inline mobile/Client/Client.Droid/Screens/ScheduleFragment.cs]: Relic. Should have been deleted

File: mobile/Client/Client.Droid/Screens/ScheduleFragment.cs (hunk 5 above)
Comment (Saadnajmi) [inline mobile/Client/Client.Droid/Screens/ScheduleFragment.cs]: As I shall

File: mobile/Common/Common.Common/ShowdownRESTClient.cs (hunk 7 above)
Comment (Saadnajmi) [inline mobile/Common/Common.Common/ShowdownRESTClient.cs]: I thought you forgot to delete "Async" since we haven't been using that naming convention. I can change it back

Comment (zuhairp) [inline mobile/Common/Common.Common/ShowdownRESTClient.cs]: Ah, ok. if you see I missed it somewhere, point it out
